ts/WoltLabSuite/Core/Ui/Poll/Manager/View/Abstract.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Abstract implementation for poll views.
+ *
+ * @author  Joshua Ruesweg
+ * @copyright  2001-2022 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/Poll/Manager/View/Abstract
+ * @since   5.5
+ */
+
+import * as Core from "../../../../Core";
+import AjaxRequest from "../../../../Ajax/Request";
+import { RequestOptions, ResponseData } from "../../../../Ajax/Data";
+import Manager, { PollViews } from "../Manager";
+
+export abstract class Abstract {
+  protected readonly pollManager: Manager;
+  protected button: HTMLButtonElement;
+
+  public constructor(manager: Manager) {
+    this.pollManager = manager;
+
+    this.initButton();
+  }
+
+  private apiCall(actionName: string, data?: RequestOptions): void {
+    const request = new AjaxRequest({
```

Comment:
You are rewriting the JS API from the ground up, why not take your time and throw away the old garbage `wcf\action\PollAction`? These manual invocations are a huge mess.